db/version_edit_handler.cc

```diff
@@ -261,14 +261,37 @@ Status VersionEditHandler::OnColumnFamilyAdd(VersionEdit& edit,
         cf_name.compare(kPersistentStatsColumnFamilyName) == 0;
     if (cf_options == name_to_options_.end() &&
         !is_persistent_stats_column_family) {
+      // distinguish between transient cfs (should not open) & cfs not passed in
+      // by user (error)
+      ROCKS_LOG_INFO(version_set_->db_options()->info_log,
+                     "Column family being marked as do not open in "
+                     "VersionEditHandler: %s (is_transient=%d)",
+                     cf_name.c_str(), edit.GetTransientColumnFamily());
       do_not_open_column_families_.emplace(edit.GetColumnFamily(), cf_name);
+      if (edit.GetTransientColumnFamily()) {
+        transient_column_families_.insert(edit.GetColumnFamily());
+      }
     } else {
       if (is_persistent_stats_column_family) {
         ColumnFamilyOptions cfo;
         OptimizeForPersistentStats(&cfo);
         tmp_cfd = CreateCfAndInit(cfo, edit);
       } else {
-        tmp_cfd = CreateCfAndInit(cf_options->second, edit);
+        // Check that client doesn't incorrectly open a transient cf as
+        // non-transient
+        if (cf_options->second.is_transient !=
```

Comment:
If a user opens the DB and passes in a transient CF handler, with is_transient = true, it seems we will follow this code path and create the CF?